Core/Src/thermistor.c

```diff
@@ -71,36 +71,35 @@ void StartReadTempTask(void *argument){
         osThreadTerminate(osThreadGetId());
     }
 
-    char concatenatedTempMessages[1024]; // TODO: make sure we don't concatenate past msg size, look at strncat()
-    char formattedTempMessage[20];
+    static char concatenatedTempMessages[256]; // TODO: make sure we don't concatenate past msg size, look at strncat()
     char* time;
+    static char* buffer_pos = concatenatedTempMessages;
     double voltages[NUM_TEMPERATURE_SENSORS];
 
     for (;;){
         if (newData_thermistor == 1) {
-
+            int written = 0;
             // Array of voltages passed by reference
             readTemperatureSensorVoltageFromADC(voltages);
 
             for(int i = 0; i < NUM_TEMPERATURE_SENSORS; i++) {
                 temperatures[i] = getTemperature(voltages[i]);
-
                 time = get_time();
-                strcat(concatenatedTempMessages,time);
+//                /* TODO: correlate the index "i" with the correct physical ADC channel
+//                 since the index may not align with the correct channel in the future */
+                int written = sprintf(buffer_pos, "[%s] ADC %d %.5f \tTemperature: %f\r\n", time, i, voltages[i], temperatures[i]);
```

Comment:
double declaration of written, could just be an int written outside the for loop:
```suggestion
                written = sprintf(buffer_pos, "[%s] ADC %d %.5f \tTemperature: %f\r\n", time, i, voltages[i], temperatures[i]);
```